### PR TITLE
MGMT-1937 - Adapt validation messages to DHCP mode

### PR DESCRIPTION
**Pull request by ori-amizur**
/assign @avishayt 
/cc @ronniel1 
/cc @YuviGold

### CODE REVIEW

File: internal/cluster/transition_test.go [hunk 1]
```diff
@@ -870,7 +870,7 @@ var _ = Describe("Refresh Cluster - Advanced networking validations", func() {
 					SufficientMastersCount:              {status: ValidationSuccess, messagePattern: "Cluster has sufficient number of master candidates"},
 					isClusterCidrDefined:                {status: ValidationFailure, messagePattern: "Cluster Network CIDR is undefined"},
 					isServiceCidrDefined:                {status: ValidationFailure, messagePattern: "Service Network CIDR is undefined"},
-					noCidrOverlapping:                   {status: ValidationPending, messagePattern: "At least one of the CIDRS .MachineNetworkCIDR, ClusterNetworkCIDR, ServiceNetworkCKIDR. is undefined"},
+					noCidrOverlapping:                   {status: ValidationPending, messagePattern: "At least one of the CIDRS .MachineNetworkCIDR, ClusterNetworkCIDR, ServiceNetworkCIDR. is undefined"},
```
Comment (avishayt) [inline internal/cluster/transition_test.go:873]:
```suggestion
					noCidrOverlapping:                   {status: ValidationPending, messagePattern: "At least one of the CIDRs (Machine Network, Cluster Network, Service Network) is undefined"},
```

File: internal/cluster/transition_test.go [hunk 2]
```diff
@@ -1061,9 +1061,9 @@ var _ = Describe("Refresh Cluster - With DHCP", func() {
 				validationsChecker: makeJsonChecker(map[validationID]validationCheckResult{
 					IsMachineCidrDefined:                {status: ValidationFailure, messagePattern: "Machine network CIDR is undefined"},
 					isMachineCidrEqualsToCalculatedCidr: {status: ValidationPending, messagePattern: "Machine network CIDR or API vip or Ingress vip is undefined"},
```
Comment (avishayt) [inline internal/cluster/transition_test.go:1063]:
```suggestion
					isMachineCidrEqualsToCalculatedCidr: {status: ValidationPending, messagePattern: "Machine network CIDR or API VIP or Ingress VIP is undefined"},
```

File: internal/cluster/transition_test.go [hunk 3]
```diff
@@ -1061,9 +1061,9 @@ var _ = Describe("Refresh Cluster - With DHCP", func() {
...
-					isApiVipDefined:                     {status: ValidationFailure, messagePattern: "API vip is undefined"},
+					isApiVipDefined:                     {status: ValidationPending, messagePattern: "Machine network CIDR is undefined"},
 					isApiVipValid:                       {status: ValidationPending, messagePattern: "API vip is undefined"},
-					isIngressVipDefined:                 {status: ValidationFailure, messagePattern: "Ingress vip is undefined"},
+					isIngressVipDefined:                 {status: ValidationPending, messagePattern: "Machine network CIDR is undefined"},
 					isIngressVipValid:                   {status: ValidationPending, messagePattern: "Ingress vip is undefined"},
```
Comment (avishayt) [inline internal/cluster/transition_test.go:1067]:
```suggestion
					isIngressVipValid:                   {status: ValidationPending, messagePattern: "Ingress VIP is undefined"},
```

File: internal/cluster/validator.go [hunk 4]
```diff
@@ -152,13 +152,22 @@ func (v *clusterValidator) printIsMachineCidrEqualsToCalculatedCidr(context *clu
...
 }
 
 func (v *clusterValidator) printIsApiVipDefined(context *clusterPreprocessContext, status validationStatus) string {
 	switch status {
+	case ValidationPending:
+		return "Machine network CIDR is undefined"
 	case ValidationFailure:
-		return "API vip is undefined"
+		if swag.BoolValue(context.cluster.VipDhcpAllocation) {
+			return "API vip is undefined; after the machine network CIDR has been defined, the API VIP is received from DHCP lease allocation task which may take up to 2 minutes"
```
Comment (avishayt) [inline internal/cluster/validator.go:167]:
```suggestion
			return "API VIP is undefined; after the machine network CIDR has been defined, the API VIP is received from DHCP lease allocation task which may take up to 2 minutes"
```

File: internal/cluster/validator.go [hunk 5]
```diff
@@ -152,13 +152,22 @@ func (v *clusterValidator) printIsMachineCidrEqualsToCalculatedCidr(context *clu
...
+		} else {
+			return "API vip is undefined; the API VIP should be filled by the end user"
```
Comment (avishayt) [inline internal/cluster/validator.go:169]:
```suggestion
			return "API VIP is undefined and must be provided"
```

File: internal/cluster/validator.go [hunk 6]
```diff
@@ -189,13 +198,22 @@ func (v *clusterValidator) printIsApiVipValid(context *clusterPreprocessContext,
...
 }
 
 func (v *clusterValidator) printIsIngressVipDefined(context *clusterPreprocessContext, status validationStatus) string {
 	switch status {
+	case ValidationPending:
+		return "Machine network CIDR is undefined"
 	case ValidationFailure:
-		return "Ingress vip is undefined"
+		if swag.BoolValue(context.cluster.VipDhcpAllocation) {
+			return "Ingress vip is undefined; after the machine network CIDR has been defined, the Ingress VIP is received from DHCP lease allocation task which may take up to 2 minutes"
```
Comment (avishayt) [inline internal/cluster/validator.go:213]:
```suggestion
			return "Ingress VIP is undefined; after the machine network CIDR has been defined, the Ingress VIP is received from DHCP lease allocation task which may take up to 2 minutes"
```

File: internal/cluster/validator.go [hunk 7]
```diff
@@ -189,13 +198,22 @@ func (v *clusterValidator) printIsApiVipValid(context *clusterPreprocessContext,
...
+		} else {
+			return "Ingress vip is undefined; the Ingress VIP should be filled by the end user"
```
Comment (avishayt) [inline internal/cluster/validator.go:215]:
```suggestion
			return "Ingress vip is undefined and must be provided"
```

File: internal/cluster/validator.go [hunk 8]
```diff
@@ -214,7 +232,7 @@ func (v *clusterValidator) isIngressVipValid(c *clusterPreprocessContext) valida
 func (v *clusterValidator) printIsIngressVipValid(context *clusterPreprocessContext, status validationStatus) string {
 	switch status {
 	case ValidationPending:
-		return "Machine network CIDR or API vip or Ingress vip is undefined"
+		return "Ingress vip is undefined"
```
Comment (avishayt) [inline internal/cluster/validator.go:235]:
```suggestion
		return "Ingress VIP is undefined"
```

File: internal/cluster/validator.go [hunk 9]
```diff
@@ -316,7 +334,7 @@ func (v *clusterValidator) printNoCidrsOverlapping(c *clusterPreprocessContext,
 		}
 		return ""
 	case ValidationPending:
-		return "At least one of the CIDRS [MachineNetworkCIDR, ClusterNetworkCIDR, ServiceNetworkCKIDR] is undefined"
+		return "At least one of the CIDRS [MachineNetworkCIDR, ClusterNetworkCIDR, ServiceNetworkCIDR] is undefined"
```
Comment (avishayt) [inline internal/cluster/validator.go:337]:
```suggestion
		return "At least one of the CIDRs (Machine Network, Cluster Network, Service Network) is undefined"
```

File: internal/cluster/transition_test.go (hunk 1 above)
Comment (ori-amizur) [reply internal/cluster/transition_test.go:873]: @avishayt The pattern above is regex, so . may be any character.  It is part of a test

**Review by eranco74 — COMMENTED** [review]
/lgtm

**Review by avishayt — APPROVED** [review]

**Pull request merged**